File: POS System - Retaurant - 2022/POS System/CustomerReports.cs
using System;
using System.IO;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace POS_System
{

    public partial class CustomerReports
    {
        private string currentpath = Directory.GetCurrentDirectory();

        public CustomerReports()
        {
            InitializeComponent();
        }

        private void cmdPreview_Click(object sender, EventArgs e)
        {

            if (rdAllCust.Checked == true)
            {
                mdlGeneral.RptUser = "ALL";
            }
            if (rdSelCust.Checked == true)
            {
                if (string.IsNullOrEmpty(cmbCustName.Text))
                {
                    mdlGeneral.gblMessage = "Invalid customer selection.";
                    My.MyProject.Forms.MsgStatus.ShowDialog(this);
                    return;
                }
                mdlGeneral.RptUser = Strings.Mid(cmbCustName.Text.Trim(), 1, 4);
            }
            mdlGeneral.FromDate = DateFrom.Value;
            if (rdDetail.Checked == true)
            {
                mdlGeneral.gblReportOption = "DETAIL";
                mdlGeneral.gblReportName = currentpath + @"\" + @"Reports\CustomerDetail.rpt";
            }
            if (rdLedger.Checked == true)
            {
                mdlGeneral.gblReportOption = "LEDGER";
                mdlGeneral.gblReportName = currentpath + @"\" + @"Reports\CustomerLedger.rpt";
            }
            My.MyProject.Forms.Reports.MdiParent = My.MyProject.Forms.Main;
            My.MyProject.Forms.Reports.Show();

        }

        private void cmdClose_Click(object sender, EventArgs e)
        {
            mdlGeneral.SqlDataSet.Clear();
            Close();
        }

        private void frmCustomerReports_Load(object sender, EventArgs e)
        {
            mdlGeneral.GetConnectionString();
            rdAllCust.Checked = true;
            rdDetail.Checked = true;
            string FillCombo = "SELECT * FROM CUSTOMERS ORDER BY CUST_CODE";
            mdlGeneral.ExecuteAdapterQuery(mdlGeneral.SqlDataAdapter, FillCombo, mdlGeneral.Connection);
            var Reader = mdlGeneral.SqlDataAdapter.SelectCommand.ExecuteReader();
            if (Reader.HasRows)
            {
                while (Reader.Read())
                    cmbCustName.Items.Add(Reader["CUST_CODE"].ToString() + "-" + Reader["CUST_NAME"].ToString());
            }
        }

        private void rdSelCust_MouseClick(object sender, MouseEventArgs e)
        {
            if (rdSelCust.Checked == true)
            {
                cmbCustName.Enabled = true;
            }
            else
            {
                cmbCustName.Enabled = false;
            }
        }

        private void rdAllCust_MouseClick(object sender, MouseEventArgs e)
        {
            if (rdAllCust.Checked == true)
            {
                cmbCustName.Enabled = false;
            }
            else
            {
                cmbCustName.Enabled = true;
            }
        }
    }
}

File: POS System - Retaurant/Backup/POS System/CustomerReports.vb
Imports System.Data.SqlClient
Imports System.IO

Public Class CustomerReports
    Dim currentpath As String = Directory.GetCurrentDirectory()

    Private Sub cmdPreview_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdPreview.Click

        If rdAllCust.Checked = True Then
            RptUser = "ALL"
        End If
        If rdSelCust.Checked = True Then
            If cmbCustName.Text = String.Empty Then gblMessage = "Invalid customer selection." : MsgStatus.ShowDialog(Me) : Exit Sub
            RptUser = Mid(cmbCustName.Text.Trim, 1, 4)
        End If
        FromDate = DateFrom.Value
        If rdDetail.Checked = True Then gblReportOption = "DETAIL" : gblReportName = currentpath & "\" & "Reports\CustomerDetail.rpt"
        If rdLedger.Checked = True Then gblReportOption = "LEDGER" : gblReportName = currentpath & "\" & "Reports\CustomerLedger.rpt"
        Reports.MdiParent = Main
        Reports.Show()

    End Sub

    Private Sub cmdClose_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdClose.Click
        SqlDataSet.Clear()
        Me.Close()
    End Sub

    Private Sub frmCustomerReports_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Call GetConnectionString()
        rdAllCust.Checked = True
        rdDetail.Checked = True
        Dim FillCombo As String = "SELECT * FROM CUSTOMERS ORDER BY CUST_CODE"
        ExecuteAdapterQuery(SqlDataAdapter, FillCombo, Connection)
        Dim Reader As SqlDataReader = SqlDataAdapter.SelectCommand.ExecuteReader
        If Reader.HasRows Then
            While Reader.Read
                cmbCustName.Items.Add(Reader.Item("CUST_CODE").ToString & "-" & Reader.Item("CUST_NAME").ToString)
            End While
        End If
    End Sub

    Private Sub rdSelCust_MouseClick(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles rdSelCust.MouseClick
        If rdSelCust.Checked = True Then
            cmbCustName.Enabled = True
        Else
            cmbCustName.Enabled = False
        End If
    End Sub

    Private Sub rdAllCust_MouseClick(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles rdAllCust.MouseClick
        If rdAllCust.Checked = True Then
            cmbCustName.Enabled = False
        Else
            cmbCustName.Enabled = True
        End If
    End Sub
End Class